cpu/nrf52/periph/i2c.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_nrf52
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation
+ *
+ * @author      Dimitri Nahm <dimitri.nahm@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "assert.h"
+#include "periph/i2c.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#ifdef I2C_NUMOF
+
+/**
+ * @brief   If any of the 4 lower bits are set, the speed value is invalid
+ */
+#define INVALID_SPEED_MASK  (0x0f)
+
+/**
+ * @brief   Initialized bus locks (we have a maximum of two devices...)
+ */
+static mutex_t locks[] = {
```

Comment:
Please use `I2C_NUMOF` (if it complicates the code the initialization can be done in the  `.._init` function) since a board might be interested in using only one.